### DISCUSSION TITLE
GPIO and sensors

### DISCUSSION THREAD

**Discussion by bickes-cgn** [bareboat-necessities/lysmarine_gen#261] [General]
Hello,
I could not find a solution to integrate sensors connected via GPIO, like revolution counters with reed sensors i.e.
Is there an easy way for non-programmers?

Regards

Holger

**Comment by mgrouch** [comment]
You can use https://github.com/SignalK/SensESP

**Reply by bickes-cgn** [reply]
Thanks @mgrouch. Right, that´s an option. Is there a wired one as well?
(That´s how it´s working now).

**Reply by mgrouch** [reply]
You better check with SensESP team how they 
support wired Ethernet and serial outputs from sensors. I believe it should support it. 
SensESP project home page is 

https://github.com/SignalK/SensESP

**Reply by mgrouch** [reply]
I initially had preference for wired sensors. 
That changed quickly. Usually your sensors are all over different places and having them wired with data wires to central boat computer asks for many wires. Look, your car tire pressure sensor in each wheel is wireless. 
Works fine. So do not be afraid of wireless sensors.

**Reply by bickes-cgn** [reply]
Hi @mgrouch, I understand your point. As I am testing now BBN as an alternative to a working Openplotter installation, I wanted to avoid SensESP, as my wired solution is working.
I will try all the rest first and if everthing is fine and I want to switch to BBN, it´s not a big thing to set up an Sensor with SensESP. 
Thanks
